librz/main/rz-diff.c

```diff
@@ -1395,6 +1395,659 @@ static bool rz_diff_unified_files(DiffContext *ctx) {
+		return "lightgray";
+	} else if (pair->similarity >= RZ_ANALYSIS_SIMILARITY_THRESHOLD) {
+		return "yellow";
+	}
+	return "red";
+}
+
+static void graphviz_dot_nodes(RzCore *core_a, RzAnalysisFunction *fcn_a, RzCore *core_b, RzAnalysisFunction *fcn_b, RzAnalysisMatchResult *result) {
+	char addr_a[32], addr_b[32];
+
+	RzAnalysisMatchPair *pair = NULL;
+	RzAnalysisBlock *bbi = NULL;
+	RzListIter *iter = NULL;
+	const char *font = get_config_or_default(core_a, "graph.font", "Courier");
+
+	rz_strf(addr_a, "0x%08" PFMT64x, fcn_a->addr);
+	rz_strf(addr_b, "0x%08" PFMT64x, fcn_b->addr);
+
+	const char *norig = fcn_a->name ? fcn_a->name : addr_a;
+	const char *nmodi = fcn_b->name ? fcn_b->name : addr_b;
+
+	// we add all the matching basic block first
+	rz_list_foreach (result->matches, iter, pair) {
+		bbi = (RzAnalysisBlock *)pair->pair_a;
+
+		const char *fillcolor = pair_color(pair);
+		char *original = basic_block_opcodes(core_a, bbi);
+		if (!original) {
+			break;
+		}
+
+		if (pair->similarity >= RZ_ANALYSIS_SIMILARITY_THRESHOLD) {
+			// if they are similar then we diff the opcodes.
+
+			char *modified = basic_block_opcodes(core_b, (RzAnalysisBlock *)pair->pair_b);
+			if (modified && strcmp(original, modified) != 0) {
+				RzDiff *dff = rz_diff_lines_new(original, modified, NULL);
+				char *diffstr = rz_diff_unified_text(dff, norig, nmodi, false, false);
+				rz_diff_free(dff);
+				free(modified);
+
+				rz_str_replace_char(diffstr, '"', '\'');
+				diffstr = rz_str_replace(diffstr, "\n", "\\l", 1);
+				rz_cons_printf("\t\"0x%08" PFMT64x "\" [fillcolor=\"%s\","
+					       "color=\"black\", fontname=\"%s\","
+					       " label=\"%s\", URL=\"%s/0x%08" PFMT64x "\"]\n",
+					bbi->addr, fillcolor, font, diffstr, fcn_a->name,
+					bbi->addr);
+				free(diffstr);
+				continue;
+			}
+
+			// sometimes the mismatch is on a call value
+			// but the output is actually the same due sym./imp.
+			// thus we ignore the similarity check and consider this
+			// as a perfect match.
+			free(modified);
+		}
+
+		rz_str_replace_char(original, '"', '\'');
+		original = rz_str_replace(original, "\n", "\\l", 1);
+		rz_cons_printf("\t\"0x%08" PFMT64x "\" [fillcolor=\"%s\","
+			       "color=\"black\", fontname=\"%s\","
+			       " label=\"%s\", URL=\"%s/0x%08" PFMT64x "\"]\n",
+			bbi->addr, fillcolor, font, original, fcn_a->name, bbi->addr);
+		free(original);
+	}
+
+	// we then add all the unmatched basic blocks
+	rz_list_foreach (result->unmatch_a, iter, bbi) {
+		char *opcodes = basic_block_opcodes(core_a, bbi);
+		if (!opcodes) {
+			break;
+		}
+
+		rz_str_replace_char(opcodes, '"', '\'');
+		opcodes = rz_str_replace(opcodes, "\n", "\\l", 1);
+		rz_cons_printf("\t\"0x%08" PFMT64x "\" [fillcolor=\"white\","
+			       "color=\"black\", fontname=\"%s\","
+			       " label=\"%s\", URL=\"%s/0x%08" PFMT64x "\"]\n",
+			bbi->addr, font, opcodes, fcn_a->name, bbi->addr);
+		free(opcodes);
+	}
+}
+
+#define PAL_JUMP "#0037da"
```

Comment:
We should take these values from config variables/color theme

Reply:
these were hardcoded also before. i will refactor them later in another PR